_studio/mfx_lib/shared/src/libmfxsw_encode.cpp

```diff
@@ -72,6 +72,12 @@ struct CodecKey {
 
     CodecKey(mfxU32 codecId, bool fei) : codecId(codecId), fei(fei) {}
 
+    enum {
+        // special value for codecId to denote plugin, it must be
+        // different from other MFX_CODEC_* used in codecId2Handlers
+        MFX_CODEC_DUMMY_FOR_PLUGIN = 0
+    };
+
```

Comment:
"Exact ordering rule is unsignificant" - this becomes untrue after this commit!! change the comment - see our discussion below.